### PR TITLE
enhancement: added mode toggle button

### LINKED ISSUE #194: 💡[Feature]: Dark Light mode
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

I would like to add a light /dark mode button to toggle between modes . The button will be on the navbar

### Use Case

It would enhance user experience

### Benefits

_No response_

### Add ScreenShots

_No response_

### Priority

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I want to work on this issue

### PR DESCRIPTION
**Pull request by Jagpreet153**
### Related Issue(s)

- Fixes #194

### Description
I have added dark /light mode toggle button 

### Changes Made

Added mui dependency , changes css of pages files , header.tsx

### Screenshots (if applicable)
Screenshot 2024-06-29 165139
image
Screenshot 2024-06-29 165211

### Checklist

Before submitting your pull request, ensure that you have completed the following tasks:

- [x] Code follows the project's style and conventions.
- [x] Code has been tested thoroughly, and all tests pass successfully.
- [x] Documentation has been updated to reflect any relevant changes.
- [x] All new and modified code is adequately commented for clarity.
- [x] Pull request title follows the specified format: [Feature/Enhancement/Bug Fix]: Brief description of the change.

### Additional Notes
Kindly review this PR .

### CODE REVIEW

**Comment by Smitkhobragade** [comment]
@Jagpreet153 can you make a Video of all where you applied the Light and dark Mode ?

**Comment by Jagpreet153** [comment]
Sure


https://github.com/oxiton-foundation/click-metrics/assets/119691733/a7235c53-1064-4dbf-b186-08f1d4702573

**Comment by Smitkhobragade** [comment]
image
In this Photo. the Black Area which I have highlighted using gray pen. can you cover it with the same background. like this kind of doesnt looks nice.
Also this is happening in the next some Screens I have attached.
image
image

In the Photo below. The Sidebar looks kind of disconnected. can you manage the color or it so it looks good. make the highlighted area of the same color as of the navbar.
image


and the last. I think here it is al white. doesn't looks like dark mode. pls solve this also.
image


@Jagpreet153 . Please Note the above changes and let me know once completed ...

**Comment by Smitkhobragade** [comment]
@Jagpreet153 Any Updates ??

**Comment by Jagpreet153** [comment]
I have updated changes
Screenshot 2024-07-02 115414
Screenshot 2024-07-02 115508
Screenshot 2024-07-02 115521
Screenshot 2024-07-02 115545
Screenshot 2024-07-02 115603

**Comment by Smitkhobragade** [comment]
Great it looks good now.

**Comment by Smitkhobragade** [comment]
@Swarnendu0123 . I have reviewed the commits. Everything looks fine. You can review and merge it.

**Comment by Swarnendu0123** [comment]
The dark mode we are targeting should be dark like GitHub. I don't like e the blue theme. Please change it to dark as github dark mode.

**Comment by Swarnendu0123** [comment]
remove the `client/package-lock.json` other we can encounter some bug in production.

**Comment by Swarnendu0123** [comment]
Attach video.

**Comment by Jagpreet153** [comment]
https://github.com/oxiton-foundation/click-metrics/assets/119691733/628eec44-73c4-408e-8cfd-e51108832fa8

File: client/package.json [hunk 1]
```diff
@@ -10,6 +10,10 @@
     "preview": "vite preview"
   },
   "dependencies": {
+    "@emotion/react": "^11.11.4",
```
Comment (Swarnendu0123) [inline client/package.json:13]: Don't add another Library just for an icon.

**Comment by Swarnendu0123** [comment]
> 
> https://github.com/oxiton-foundation/click-metrics/assets/119691733/628eec44-73c4-408e-8cfd-e51108832fa8
> 
> 

In this video, I can see the work is very unstructured. You haven't taken care of borders. The search boxes, the entire Analytics page is white. Also the colour I have talked about is not black, it's a dark grey type colour.

**Comment by Jagpreet153** [comment]
@Swarnendu0123 Ok I will makes the color to dark gray

File: client/package.json (hunk 1 above)
Comment (Jagpreet153) [reply client/package.json:13]: This icon is the day night icon for dark / light mode

Comment (Smitkhobragade) [reply client/package.json:13]: @Swarnendu0123 is there any library we should use for icon ?

**Pull request closed without merging**